h/views/api_flags.py

```diff
@@ -0,0 +1,41 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import unicode_literals
+
+from pyramid import security
+from pyramid.httpexceptions import HTTPNoContent, HTTPNotFound
+
+from memex.resources import AnnotationResourceFactory
+from memex.views import api_config
+
+from h.schemas import ValidationError
+
+
+@api_config(route_name='api.flags',
+            request_method='POST',
+            link_name='flags.create',
+            description='Flag an annotation for review.',
+            effective_principals=security.Authenticated)
+def create(request):
+    annotation = _fetch_annotation(request)
+    svc = request.find_service(name='flag')
+    svc.create(request.authenticated_user, annotation)
+    return HTTPNoContent()
+
+
+def _fetch_annotation(request):
+    annotation_id = request.json_body.get('annotation')
+    if not annotation_id:
+        raise ValidationError('annotation id is required')
+
+    not_found_msg = 'cannot find annotation with id %s' % annotation_id
+
+    try:
+        factory = AnnotationResourceFactory(request)
```

Comment:
Rather than doing this here, I think it might be cleaner to leave the route with `factory='memex.resources. AnnotationResourceFactory'` but just omit the `traverse=` argument. Then in the view you can get the annotation as `resource = context[annotation_id]`.